### PR TITLE
ci(.github): modify helm-release to allow inheriting action

### PR DESCRIPTION
**Pull request by lahabana**
This will make it easier for child repo to reuse this

Signed-off-by: Charly Molter <charly.molter@konghq.com>

### Checklist prior to review

- [x] Link to docs PR or issue --
- [x] Link to UI issue or PR --
- [x] Is the [issue worked on linked][1]? --
- [x] The PR does not hardcode values that might break projects that depend on kuma (e.g. "kumahq" as a image registry) --
- [x] The PR will work for both Linux and Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Unit Tests --
- [x] E2E Tests --
- [x] Manual Universal Tests --
- [x] Manual Kubernetes Tests --
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: .github/workflows/helm-release.yaml [hunk 1]
```diff
@@ -46,8 +47,8 @@ jobs:
           curl -L "https://github.com/helm/chart-releaser/releases/download/v${CR_VERSION}/chart-releaser_${CR_VERSION}_linux_amd64.tar.gz" | sudo tar xvz --directory /usr/bin cr
 
           make dev/tools
-          git checkout go.*
       - name: Update chart versions
+        id: update-chart-version
```
Comment (slonka) [inline .github/workflows/helm-release.yaml:51]: not used anywhere but I guess it does not hurt

File: .github/workflows/helm-release.yaml [hunk 2]
```diff
@@ -56,19 +57,19 @@ jobs:
...
-
-          ./tools/releases/helm.sh ${args}
-
+          ${{ env.KUMA_DIR }}/tools/releases/helm.sh ${args}
+      - name: Commit chart versions
+        run: |
           git add -u deployments/charts
           # This commit never ends up in the repo
-          git commit -m "ci(helm): update versions"
+          git commit --allow-empty -m "ci(helm): update versions"
```
Comment (slonka) [inline .github/workflows/helm-release.yaml:65]: Why do we need to `--allow-empty`? I vaguely remember some conversation about this on slack and I'm not sure if this does not have any negative ramifications (or maybe it was dirty index and adding `-am`).

**Review by slonka — APPROVED** [review]
Have you tried running this to check if it works?

File: .github/workflows/helm-release.yaml (hunk 2 above)
Comment (lahabana) [reply .github/workflows/helm-release.yaml:65]: KM has allow-empty and this didn't. Had to pick one and picked adding `--allow-empty` as mentioned this commit is not actually pushed anywhere so I think it doesn't matter (and avoids failing git commit)

File: .github/workflows/helm-release.yaml (hunk 1 above)
Comment (lahabana) [reply .github/workflows/helm-release.yaml:51]: It's actually used in my matching PR in KM

**Comment by lahabana** [comment]
> Have you tried running this to check if it works?
Yep this one runs as a check on PR: https://github.com/kumahq/kuma/actions/runs/4063943391/jobs/6996770748

**Pull request merged**